pytensor/tensor/rewriting/linalg.py

```diff
@@ -964,3 +965,58 @@ def rewrite_cholesky_diag_to_sqrt_diag(fgraph, node):
             non_eye_input = pt.shape_padaxis(non_eye_input, -2)
 
     return [eye_input * (non_eye_input**0.5)]
+
+
+# SLogDet Rewrites
+def check_sign_det(node):
+    if not (isinstance(node.op, Elemwise) and isinstance(node.op.scalar_op, Sign)):
+        return False
+
+    return True
+
+
+def check_log_abs_det(node):
+    if not (isinstance(node.op, Elemwise) and isinstance(node.op.scalar_op, Log)):
+        return False
+
+    potential_abs = node.inputs[0].owner
+    if not (
+        isinstance(potential_abs.op, Elemwise)
+        and isinstance(potential_abs.op.scalar_op, Abs)
+    ):
+        return False
+
+    return True
+
+
+def check_log_det(node):
+    if not (isinstance(node.op, Elemwise) and isinstance(node.op.scalar_op, Log)):
+        return False
+
+    return True
+
+
+@node_rewriter(tracks=[det])
+def slogdet_specialization(fgraph, node):
+    x = node.inputs[0]
+    sign_det_x, slog_det_x = SLogDet()(x)
+    replacements = {}
+    for client in list(fgraph.clients.keys()):
+        # Check for sign(det)
+        if check_sign_det(client[0].owner):
+            replacements[client[0].owner.outputs[0]] = sign_det_x
+
+        # Check for log(abs(det))
+        elif check_log_abs_det(client[0].owner):
+            replacements[client[0].owner.outputs[0]] = slog_det_x
```

Comment:
This won't work, because the helper checks for log first and then abs, but the direct client of `det` will be `abs` not `log`. `log` will itself be the client of `abs`